PWGMM/Tasks/uecharged.cxx

```diff
@@ -0,0 +1,336 @@
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+/// \author Antonio Ortiz (antonio.ortiz.velasquez@cern.ch)
+/// \since November 2021
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "Common/DataModel/Multiplicity.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/Core/TrackSelectionDefaults.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Framework/HistogramRegistry.h"
+#include <TH1F.h>
+#include <TH2F.h>
+#include <TF1.h>
+#include <TRandom.h>
+#include <cmath>
+
+// TODO: implement flags for all, phys. sel., good vertex, ...
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+struct uecharged {
+
+  TrackSelection globalTrackswoPrim; // Track without cut for primaries
+  TrackSelection globalTracks;       // Track with cut for primaries
+
+  HistogramRegistry ue{"ue", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+  static constexpr std::string_view pNumDenMeasuredPS[3] = {"pNumDenMeasuredPS_NS", "pNumDenMeasuredPS_AS", "pNumDenMeasuredPS_TS"};
+  static constexpr std::string_view pSumPtMeasuredPS[3] = {"pSumPtMeasuredPS_NS", "pSumPtMeasuredPS_AS", "pSumPtMeasuredPS_TS"};
+  static constexpr std::string_view hPhi[3] = {"hPhi_NS", "hPhi_AS", "hPhi_TS"};
+  // data driven correction
+  static constexpr std::string_view hNumDenMCDd[3] = {"hNumDenMCDd_NS", "hNumDenMCDd_AS", "hNumDenMCDd_TS"};
+  static constexpr std::string_view hSumPtMCDd[3] = {"hSumPtMCDd_NS", "hSumPtMCDd_AS", "hSumPtMCDd_TS"};
+  static constexpr std::string_view hNumDenMCMatchDd[3] = {"hNumDenMCMatchDd_NS", "hNumDenMCMatchDd_AS", "hNumDenMCMatchDd_TS"};
+  static constexpr std::string_view hSumPtMCMatchDd[3] = {"hSumPtMCMatchDd_NS", "hSumPtMCMatchDd_AS", "hSumPtMCMatchDd_TS"};
+  // hist data for corrections
+  static constexpr std::string_view hPtVsPtLeadingData[3] = {"hPtVsPtLeadingData_NS", "hPtVsPtLeadingData_AS", "hPtVsPtLeadingData_TS"};
+  static constexpr std::string_view pNumDenData[3] = {"pNumDenData_NS", "pNumDenData_AS", "pNumDenData_TS"};
+  static constexpr std::string_view pSumPtData[3] = {"pSumPtData_NS", "pSumPtData_AS", "pSumPtData_TS"};
+
+  OutputObj<TF1> f_Eff{"fpara"};
+
+  void init(o2::framework::InitContext&)
+  {
+    // primaries w/o golden cut
+    globalTracks = getGlobalTrackSelection();
+    globalTracks.SetRequireGoldenChi2(false);
+    // all w/o golden cuts
+    globalTrackswoPrim = getGlobalTrackSelection();
+    globalTrackswoPrim.SetMaxDcaXYPtDep([](float pt) { return 3.f + pt; });
+    globalTrackswoPrim.SetRequireGoldenChi2(false);
+
+    std::vector<double> ptBinningL = {
```

Comment:
Note that binning can be also supplied with `ConfigurableAxis` (there are examples in the tutorials and some tasks already use it).